slowquant/qiskit_interface/custom_ansatz.py

```diff
@@ -21,13 +27,16 @@ def tUPS(
     mapper: FermionicMapper,
     ansatz_options: dict[str, Any],
 ) -> tuple[QuantumCircuit, dict[str, int]]:
-    """tUPS ansatz.
+    """Create tUPS ansatz.
 
-    #. 10.48550/arXiv.2312.09761
+    #. 10.1103/PhysRevResearch.6.023300
+    #. 10.1088/1367-2630/ac2cb3 (QNP)
 
     Ansatz Options:
         * n_layers [int]: Number of layers.
-        * do_pp [bool]: Do perfect pairing.
+        * do_pp [bool]: Do perfect pairing. (default: False)
+        * do_qnp [bool]: Do QNP tiling. (default: False)
```

Comment:
Having QNP within tUPS is a bit weird. 
Would a separate function not be nicer? Especially in case we do the 2nd order one day.

Reply:
Right now the two code-branches QNP and tUPS are almost identical.
All improvements/options I might want to add to tUPS are equally relevant for QNP.
Having them split is very tedious when adding new options.